server/s3_request_object.cc

```diff
@@ -171,9 +172,37 @@ void S3RequestObject::populate_and_log_audit_info() {
 
   // Skip audit logs for health checks.
   if (audit_log_obj.get_publish_flag()) {
-  if (S3AuditInfoLogger::save_msg(request_id, audit_log_obj.to_string()) < 0) {
-    s3_log(S3_LOG_FATAL, request_id, "Audit Logger Error. STOP Server");
-  }
+    if (S3AuditInfoLogger::save_msg(request_id, audit_log_obj.to_string()) <
+        0) {
+      s3_log(S3_LOG_FATAL, request_id, "Audit Logger Error. STOP Server");
+    }
   }
   s3_log(S3_LOG_DEBUG, request_id, "Exiting");
 }
+
+bool S3RequestObject::validate_attrs(const std::string& c_bucket_name,
+                                     const std::string& c_object_name) {
+  // NOTE: don't call get_object_name() and get_bucket_name() twice;
+  // UTs have expectations about amount and order of these calls.
+  const std::string& req_object_name = get_object_name();
+  const std::string& req_bucket_name = get_bucket_name();
+
+  namespace s3cu = S3CommonUtilities;
+
+  if (s3_fi_is_enabled("di_metadata_bucket_or_object_corrupted") ||
+      req_bucket_name != c_bucket_name || req_object_name != c_object_name) {
+    s3_iem(LOG_ERR, S3_IEM_OBJECT_METADATA_NOT_VALID,
+           S3_IEM_OBJECT_METADATA_NOT_VALID_STR,
+           S3_IEM_OBJECT_METADATA_NOT_VALID_JSON,
+           !s3cu::isprints(req_bucket_name) ? "@@@corrupted@@@"
+                                            : req_bucket_name.c_str(),
+           !s3cu::isprints(c_bucket_name) ? "@@@corrupted@@@"
+                                          : c_bucket_name.c_str(),
+           !s3cu::isprints(req_object_name) ? "@@@corrupted@@@"
+                                            : req_object_name.c_str(),
+           !s3cu::isprints(c_object_name) ? "@@@corrupted@@@"
+                                          : c_object_name.c_str());
```

Comment:
Let's do this the following way:

bool isprints(const std::string &attr) {
  return std::all_of(attr.begin(), attr.end(), [](std::string::value_type c) {
    return static_cast<bool>(std::isprint(c),  **std::locale("")**);
  });
}

Initially isprints was prepared for unicode (it can be seen by std::string::value_type) still I forgot that unicode version of isprint uses locale info.

Reply:
std::locale("") was added, but this is "User-preferred locale setting" and on server side. is it enough?

Reply:
Latest fixes remove isprints completely, which is in alignment with the rest of the S3 code.  Resolving this comment.